src/azul/service/manifest_service.py

```diff
@@ -703,10 +704,55 @@ def write_to(self, output: IO[str]) -> Optional[str]:
             url = furl(config.service_endpoint(),
                        path=f'/repository/files/{uuid}',
                        args=dict(version=version, catalog=self.catalog))
+            name = self._validate_path(name)
             output.write(f'url={self._option(url.url)}\n'
                          f'output={self._option(name)}\n\n')
         return None
 
+    @classmethod
+    def _validate_path(cls, path: str) -> str:
+        """
+        >>> f = CurlManifestGenerator._validate_path
+        >>> f('foo/bar/\\x1F/file')
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: Invalid file name 'foo/bar/\\x1f/file'
+
+        >>> f('foo/bar/\\/file')
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: Invalid file name 'foo/bar/\\\\/file'
+
+        >>> f('foo/bar/COM6/file')
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: Invalid file name 'foo/bar/COM6/file'
+
+        >>> f('foo/bar/ / baz/file')
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: Invalid file name 'foo/bar/ / baz/file'
+
+        >>> f('foo/bar?/fi*le|')
+        'foo/bar_/fi_le_'
+
+        >>> f('foo/bar/file.fastqgz')
+        'foo/bar/file.fastqgz'
+        """
+        msg = f'Invalid file name {path!r}'
+        for character in path:
+            reject(character == u'\\' or unicodedata.category(character) == 'Cc', msg)
+        path = re.sub(r'[<>:"|?*]', '_', path)
+        path = re.fullmatch(r'(\.?[^./ ])([^/]*[^./ ])?(/(\.?[^./ ])([^/]*[^./ ])?)*',
```

Comment:
We should explain what this does. Use https://docs.python.org/3/library/re.html#re.VERBOSE

Reply:
why do you assign to path? fullmatch returns a match object. we just want to know if the re matches.

Reply:
I was under the impression that it bare more validity if the path was obtained from the regex object, since it was a match, and because I was under the impression you might prefer it that way.

Reply:
> I was under the impression that it bare more validity 

Validity is a binary concept. Something is either valid or not.

Under what circumstances would `re.fullmatch(pattern, input).string != input`?

> I was under the impression you might prefer it that way.

I don't prefer convolution.